### PR TITLE
#195 added config 'identityAttribute'

### PR DESCRIPTION
**Pull request by rochamarcelo**
for issue #195

### CODE REVIEW

File: tests/TestCase/AuthenticationServiceTest.php [hunk 1]
```diff
@@ -241,6 +308,103 @@ public function testPersistIdentity()
...
+            'florian',
+            $result['request']->getAttribute('session')->read('Auth.username')
+        );
+
+        $identity = $result['request']->getAttribute('customIdentity');
+        $this->assertInstanceOf(IdentityInterface::class, $identity);
+        $this->assertEquals($data, $identity->getOriginalData());
+
+        $data = ['username' => 'johndoe'];
+        $this->assertEquals($data, $result['request']->getAttribute('identity'));
```
Comment (markstory) [inline tests/TestCase/AuthenticationServiceTest.php]: Why is this attribute set if there is a custom attribute in use.

Comment (markstory) [reply tests/TestCase/AuthenticationServiceTest.php]: Ok.

**Comment by robertpustulka** [comment]
Could you add a similar config key to the middleware and the component?

**Comment by rochamarcelo** [comment]
For middleware I would need to update the constructor, is it a good idea? If the getConfig method was defined at AuthenticationServiceInterface I could do 'service->getConfig('identityAttribute'); . What approach do you recommend?

**Comment by robertpustulka** [comment]
@rochamarcelo I don't think exposing `getConfig` is a good idea.

For the middleware I suppose we could still change the second argument to be a config array and use `InstanceConfigTrait` (similar to the authz plugin middleware).

This could be done in a BC way:
```php
public function __construct($subject, $config = [])
{
    if (is_string($config)) {
        $config = ['name' => $config];
    }
    $this->setConfig($config);
    ...
}

**Comment by rochamarcelo** [comment]
@robertpustulka @markstory how can I fix the following travis ci build?

 ------ ------------------------------------------------------------- 
  Line   src/Middleware/AuthenticationMiddleware.php                  
 ------ ------------------------------------------------------------- 
  66     Call to function is_string() will always evaluate to false.  
 ------ -------------------------------------------------------------

**Comment by rochamarcelo** [comment]
@robertpustulka thanks. I fixed the param type

File: src/AuthenticationService.php [hunk 2]
```diff
@@ -85,7 +85,8 @@ class AuthenticationService implements AuthenticationServiceInterface
     protected $_defaultConfig = [
         'authenticators' => [],
         'identifiers' => [],
-        'identityClass' => Identity::class
+        'identityClass' => Identity::class,
```
Comment (burzum) [inline src/AuthenticationService.php]: I don't think this should be here. It should be IdentityInterfacen::class and when we use the interface there is no need to make it configurable. We want to enforce the interface as a minimum common API of identities. There should be no way to bypass that.

**Pull request merged**